src/system.ts

```diff
@@ -295,10 +295,14 @@ export class System extends Element {
     this.startX = startX;
     this.debugNoteMetricsYs = debugNoteMetricsYs;
     this.lastY = y;
-    this.boundingBox = new BoundingBox(this.options.x, this.options.y, this.options.width, this.lastY - this.options.y);
     Stave.formatBegModifiers(this.partStaves);
   }
 
+  /** Get the boundingBox. */
+  getBoundingBox(): BoundingBox {
+    return new BoundingBox(this.options.x, this.options.y, this.options.width, this.lastY ?? 0 - this.options.y);
```

Comment:
0 - this.options.y

Can't it just be -this.options.y

Reply:
Oh wait... is there a operator precedence thing going on? Do we always want to subtract this.options.y?

Reply:
this.lastY is normally not undefined. So it is this.lasty - this.options.y

Reply:
Anyways, I think `this.lastY ?? 0 - this.options.y` is confusing. Not sure what the code is trying to do....

I think you're trying to say:

`this.lastY`

OR if the above is `undefined`, then 

`-this.options.y`

Correct?

Reply:
Make sure you test the operator precedence here.

Doesn't ?? just result in `this.lastY` if as you say, it is normally defined?

Reply:
I think we should add parentheses for clarity, or extract a local variable so it's doing what you actually want it to do.

Reply:
I will check right now

Reply:
Precedence is on ??. I have put parentheses to avoid confusion.